### PR TITLE
Add policy set for GitHub provider

### LINKED ISSUE #325: Rule for github_repository seems to be wrongly placed under gcp
The rule `accurics.gcp.IAM.145` is currently located under [gcp](https://github.com/accurics/terrascan/commits/master/pkg/policies/opa/rego/gcp/github_repository), which isn't intuitive as it is targeting the GitHub provider and not GCP. It would be better to move its own policy set.

### PR DESCRIPTION
**Pull request by HorizonNet**
Add GitHub as a new provider with the existing policy copied over from
the gcp policies.

Resolves #325

### CODE REVIEW

**Comment by HorizonNet** [comment]
Tested locally via `bin/terrascan scan -t github -p pkg/policies/opa/rego/github -f repositories.tf`. Violations are shown as expected. Missing point is that violations from the original gcp policy is also reported when running `bin/terrascan scan -t github -p pkg/policies/ -f repositories.tf`.

**Review by williepaul — CHANGES_REQUESTED** [review]

File: docs/policies/github.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+### github_repository
+| Category | Resource | Severity | Description | Reference ID |
+| -------- | -------- | -------- | ----------- | ------------ |
+| Identity & Access Management | gcp | MEDIUM | Repository is Not Private. | accurics.gcp.IAM.145 |
```
[the remark this answers is not in the document]
Comment (HorizonNet) [reply docs/policies/github.md]: Ah, already thought that I missed this one. Updated.

File: pkg/policies/opa/rego/github/github_repository/accurics.github.IAM.145.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "privateRepoEnabled",
+    "file": "privateRepoEnabled.rego",
+    "template_args": null,
+    "severity": "MEDIUM",
+    "description": "Repository is Not Private.",
+    "reference_id": "accurics.github.IAM.145",
```
[the remark this answers is not in the document]
Comment (HorizonNet) [reply pkg/policies/opa/rego/github/github_repository/accurics.github.IAM.145.json]: Done. Also changed the file name accordingly.

File: pkg/policies/opa/rego/github/github_repository/privateRepoEnabled.rego [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package accurics
+
+privateRepoEnabled[api.id] {
+    api := input.github_repository[_]
+    not api.config.private == true
```
[the remark this answers is not in the document]
Comment (HorizonNet) [reply pkg/policies/opa/rego/github/github_repository/privateRepoEnabled.rego:5]: It's in ;)

**Comment by HorizonNet** [comment]
@williepaul Thanks for the feedback. Addressed your points. Let me make this one ready for review as you already gave feedback.

**Review by williepaul — APPROVED** [review]

**Review by williepaul — APPROVED** [review]
@HorizonNet all looks good, I tested the changes out locally and they worked fine. We really appreciate the contribution!

**Pull request merged**